sinatra-contrib/lib/sinatra/reloader.rb

```diff
@@ -251,7 +251,11 @@ def self.perform(klass)
         watcher.update
         reloaded_paths << watcher.path
       end
-      @@after_reload.each { |block| block.call(reloaded_paths) } if reloaded_paths.any?
+      return if reloaded_paths.empty?
+      @@after_reload.each do |block|
+        args = (block.lambda? && block.arity != 1) ? [] : [reloaded_paths]
```

Comment:
nit: I guess, we can respect consistency with sinatra's one: https://github.com/sinatra/sinatra/blob/2e980f3534b680fbd79d7ec39552b4afb7675d6c/lib/sinatra/base.rb#L1681

Reply:
That's a good catch. Should this be factored into a method and used from both places? Line 256 here specifically avoids Ruby throwing an error in the case given -- when the block was passed as an arg after being defined as a lambda (not with `proc`) with no args (see my 2 comments starting at https://github.com/sinatra/sinatra/pull/1692#discussion_r631511938).